### PR TITLE
Feat: Update ITestResultSeralizer to accept messages.

### PR DESCRIPTION
**Pull request by Siphonophora**
Pass messages to seralizer as discussed in https://github.com/spekt/junit.testlogger/pull/33

### CODE REVIEW

**Comment by Siphonophora** [comment]
@codito So, this is pretty simple. I was a bit surprised by the two tests that were failing. I think those are just windows specific issues. 

Let me know if you think I should add tests to this. The part I care about for junit will be covered by the unit test that is currently failing in https://github.com/spekt/junit.testlogger/pull/33

**Review by codito — APPROVED** [review]
:+1: thanks! Will add a windows CI soon.

**Comment by codito** [comment]
It will be good to add a test. May be we can update the `JsonTestResultSerializer` output schema to include `messages`: https://github.com/spekt/testlogger/blob/master/test/TestLogger.UnitTests/TestDoubles/JsonTestResultSerializer.cs#L20, this will allow us to assert the output content in `TestRunCompleteWorkflowTests`.

**Comment by Siphonophora** [comment]
How about this as an option. Its simpler than putting messages in the JSON. Any downside to this?

**Comment by codito** [comment]
`JsonTestResultSerializer` is also part of the json test logger which is used in Acceptance Tests. With this approach, we may not be able to validate messages as part of an acceptance test (in future).

File: test/TestLogger.UnitTests/TestRunCompleteWorkflowTests.cs [hunk 1]
```diff
@@ -95,21 +95,44 @@ public void CompleteShouldSerializeTestResults()
...
-            testRun.Complete(this.testRunCompleteEvent);
+            relativePathTestRun.Complete(this.testRunCompleteEvent);
 
-            var logFilePath = testRun.LoggerConfiguration.LogFilePath;
+            var logFilePath = relativePathTestRun.LoggerConfiguration.LogFilePath;
             Assert.IsFalse(string.IsNullOrEmpty(this.fileSystem.Read(logFilePath)));
         }
 
+        [TestMethod]
+        public void CompleteShouldPassAllMessagesToSeralizer()
```
Comment (codito) [inline test/TestLogger.UnitTests/TestRunCompleteWorkflowTests.cs:114]: nit: minor typo in `Serializer` (also in line 116).

**Comment by Siphonophora** [comment]
Yep makes sense. We should be all set now. 

Note, I dropped traits from the sample json, since they aren't in the `Test` model.

**Comment by codito** [comment]
Looks perfect! Thanks.

**Pull request merged**